tests/integration_tests/test_wandb.py

```diff
@@ -31,79 +32,163 @@ def _multiobjective_func(trial: optuna.trial.Trial) -> Tuple[float, float]:
 @mock.patch("optuna.integration.wandb.wandb")
 def test_run_initialized(wandb: mock.MagicMock) -> None:
 
+    wandb.sdk.wandb_run.Run = mock.MagicMock
+
     wandb_kwargs = {
         "project": "optuna",
         "group": "summary",
         "job_type": "logging",
         "mode": "offline",
+        "tags": ["test-tag"],
     }
 
-    WeightsAndBiasesCallback(
-        metric_name="mse",
-        wandb_kwargs=wandb_kwargs,
+    WeightsAndBiasesCallback(metric_name="mse", wandb_kwargs=wandb_kwargs, as_multirun=False)
+    wandb.init.assert_called_once_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandbc = WeightsAndBiasesCallback(
+        metric_name="mse", wandb_kwargs=wandb_kwargs, as_multirun=True
     )
+    wandb.run = None
+
+    study = optuna.create_study(direction="minimize")
+    _wrapped_func = wandbc.track_in_wandb(_objective_func)
+    study.optimize(_wrapped_func, n_trials=1, callbacks=[wandbc])
+
+    wandb.init = mock.MagicMock()
+    study_id = study._study_id
+    frozen_trial = study.trials[0]
+    trial_id = study._storage.get_trial_id_from_study_id_trial_number(
+        study_id, frozen_trial.number
+    )
+    trial = optuna.Trial(study, trial_id)
+
+    _ = _wrapped_func(trial)
 
     wandb.init.assert_called_once_with(
-        project="optuna",
-        group="summary",
-        job_type="logging",
-        mode="offline",
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
+    )
+
+    wandb.init = mock.MagicMock()
+    study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
+    wandb.init.assert_called_once_with(
+        project="optuna", group="summary", job_type="logging", mode="offline", tags=["test-tag"]
     )
 
 
 @mock.patch("optuna.integration.wandb.wandb")
 def test_attributes_set_on_epoch(wandb: mock.MagicMock) -> None:
```

Comment:
General comments to this test:
* We don't have to test `track_in_wandb` decorator, as it plays no part in setting the attributes.
* For the remaining part, I think we can keep logic from `master` (modified with `enqueue_trial` and mocking correct objects, as in your test ofc) and just parametrize `as_multirun`. Behavior doesn't change.

Something like:
```python
@mock.patch("optuna.integration.wandb.wandb")
@pytest.mark.parametrize("as_multirun", [True, False])
def test_attributes_set_on_epoch(wandb: mock.MagicMock, as_multirun: bool) -> None:

    # Vanilla update
    wandb.sdk.wandb_run.Run = mock.MagicMock

    wandbc = WeightsAndBiasesCallback(as_multirun=as_multirun)
    study = optuna.create_study(direction="minimize")
    study.enqueue_trial({"x": 1.1, "y": 2.2})
    study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])

    expected = {"direction": ["MINIMIZE"], "x": 1.1, "y": 2.2}
    wandb.run.config.update.assert_called_once_with(expected)
```

*Didn't check if linters pass :^)*